apollo-router/src/plugins/telemetry/metrics/filter.rs

```diff
@@ -0,0 +1,316 @@
+                delegate,
+                deny: self.deny.clone(),
+                allow: self.allow.clone(),
+            }),
+        )
+    }
+}
+
+#[cfg(test)]
+mod test {
+    use std::collections::HashSet;
+    use std::sync::atomic::AtomicU64;
+    use std::sync::atomic::Ordering;
+    use std::sync::Arc;
+    use std::sync::Mutex;
+
+    use opentelemetry::metrics::noop;
+    use opentelemetry::metrics::Counter;
+    use opentelemetry::metrics::InstrumentProvider;
+    use opentelemetry::metrics::Meter;
+    use opentelemetry::metrics::MeterProvider;
+    use opentelemetry::metrics::Unit;
+    use opentelemetry::Context;
+    use opentelemetry::InstrumentationLibrary;
+
+    use crate::plugins::telemetry::metrics::filter::FilterMeterProvider;
+
+    #[derive(Default, Clone)]
+    struct MockInstrumentProvider {
+        #[allow(clippy::type_complexity)]
+        counters_created: Arc<Mutex<HashSet<(String, Option<String>, Option<Unit>)>>>,
+        callbacks_registered: Arc<AtomicU64>,
+    }
+
+    impl InstrumentProvider for MockInstrumentProvider {
+        // We're only going to bother with testing counters and callbacks because the code is implemented as a macro and if it's right for counters it's right for everything else.
+        fn u64_counter(
+            &self,
+            name: String,
+            description: Option<String>,
+            unit: Option<Unit>,
+        ) -> opentelemetry::metrics::Result<Counter<u64>> {
+            self.counters_created
+                .lock()
+                .expect("lock should not be poisoned")
+                .insert((name, description, unit));
+            Ok(Counter::new(Arc::new(noop::NoopSyncInstrument::new())))
+        }
+
+        fn register_callback(
+            &self,
+            _callback: Box<dyn Fn(&Context) + Send + Sync>,
+        ) -> opentelemetry::metrics::Result<()> {
+            self.callbacks_registered.fetch_add(1, Ordering::SeqCst);
+            Ok(())
+        }
+    }
+
+    #[derive(Default, Clone)]
+    struct MockMeterProvider {
+        instrument_provider: Arc<MockInstrumentProvider>,
+    }
+
+    impl MeterProvider for MockMeterProvider {
+        fn versioned_meter(
+            &self,
+            name: &'static str,
+            version: Option<&'static str>,
+            schema_url: Option<&'static str>,
+        ) -> Meter {
+            Meter::new(
+                InstrumentationLibrary::new(name, version, schema_url),
+                self.instrument_provider.clone(),
+            )
+        }
+    }
+
+    #[test]
+    fn test_apollo_metrics() {
+        let delegate = MockMeterProvider::default();
+        let filtered = FilterMeterProvider::apollo_metrics(delegate.clone())
+            .versioned_meter("filtered", None, None);
+        filtered.u64_counter("apollo.router.operations").init();
+        filtered.u64_counter("apollo.router.operation").init();
```

Comment:
I don't see `apollo.router.operation` (singular) anywhere but in this test. Is this intentional?

Reply:
removing it